lib/data/replacements.json

```
@@ -669,6 +670,18 @@
       }
     ]
   },
+  "gradle-wrapper-validation-action": {
+    "description": "gradle/wrapper-validation-action got renamed to gradle/actions/wrapper-validation",
+    "packageRules": [
+      {
+        "matchCurrentVersion": "/3(\\.\\d+\\.\\d+)?/",
```

Is the intent to allow either `3` or e.g. `3.2.10`? and not e.g. `3.2`?

---

The code started with:

```
        "matchCurrentVersion": "3",
```

I asked if that would match all possible versions in the `3` range:

- https://github.com/renovatebot/renovate/pull/31645#discussion_r1783061192

Our developers docs list the [rules for replacement presets](https://github.com/renovatebot/renovate/blob/main/docs/development/creating-editing-renovate-presets.md#rules-for-replacement-presets):

> #### Rules for replacement presets
> 
> 1. Replacement PRs should ideally propose a replacement only once the user is on a compatible version, by specifying a compatible `matchCurrentVersion` constraint
> 1. If no compatible replacement upgrade is possible, it's acceptable to propose an incompatible one (e.g. a major version upgrade)
> 1. Replacements should update the user to the first recommended version of the new dependency and not include any new changes - whether breaking or not - if they can be avoided, in short: pin the new version
> 
> If possible, replacement presets should give the user a replacement version that is _functionally identical_ to the _last version_ under the _old_ name.
> We only want a user's tests to fail because of _broken references_ to the old package name, and not because the maintainer(s) changed the _behavior_ of the package.

The key phrase in the docs is:

> Replacement PRs should ideally propose a replacement only once the user is on a compatible version, by specifying a compatible `matchCurrentVersion` constraint

So then the question is, is just being on any version 3 of the program good enough for a user to quickly merge the replacement PR? Or does the user need to be on the last version of the old package?

---

> So then the question is, is just being on any version 3 of the program good enough for a user to quickly merge the replacement PR? Or does the user need to be on the last version of the old package?

@HonkingGoose I think it's good replace the dependency when on any version 3.

---

OK but should it be `"matchCurrentVersion": "/^3\\.",` ? Note the ^ to enforce it at the start of the line, and the mandatory `.` (assumes that there will always be some precision to the version and not just `"3" itself, is that safe?)

---

The version can be `v3` or `v3.5.0`:
- https://github.com/mueller-ma/Coffee/blob/main/.github/workflows/build.yml#L27
- https://github.com/mueller-ma/PrepaidBalance/blob/992ea54c1be81ba1e3bf2843f6a41cff43b5e153/.github/workflows/build.yml#L21

So the current regex should be fine, correct?

---

> OK but should it be `"matchCurrentVersion": "/^3\\.",` ? Note the ^ to enforce it at the start of the line, and the mandatory `.` (assumes that there will always be some precision to the version and not just `"3" itself, is that safe?)

```suggestion
        "matchCurrentVersion": "/^3\\.",
```

This suggestion is the copy/pasted code from `@rarkins`. I don't know if it's a good idea to apply the suggestion...

I'll let the maintainers decide what to do with the code. 😉

---

The current one allows `3`, `30`, `3d`, etc. Is careful validation necessary to prevent these?

---

The repo for the old version has been archived, so there shouldn't be any new versions like `30`: https://github.com/gradle/wrapper-validation-action

---

But are we prevented from having _actually correct_ matching?

---

I updated the regex. It should match `3` and `3.*` where `*` stands for anything.